src/main/java/com/ufcg/sad/models/aluno/Aluno.java

```diff
@@ -98,6 +103,31 @@ public void setMatriculas(Set<Matricula> matriculas) {
         this.matriculas = matriculas;
     }
 
+    /**
+     * Adiciona uma nova Disciplina ao aluno.
+     *
+     * @param disciplina disciplina a ser adicionada.
+     */
+    public void adicionarDisciplina(Disciplina disciplina) {
+        Matricula matricula = new Matricula(this, disciplina);
+        this.disciplinas.add(matricula);
+        disciplina.getAlunos().add(matricula);
+    }
+
+    /**
+     * Remove uma disciplina
+     *
+     * @param disciplina disciplina a ser removida
+     */
+    public void removerDisciplina(Disciplina disciplina) {
```

Comment:
Em java usando esse iterator é tranquilo remover no mesmo for? (Não uso muito Java, mas sei que em algumas linguagens da errado remover enquanto itera)